### ISSUE TITLE
How to update without select?

### ISSUE THREAD

**Issue by angelvega93** [typeorm/typeorm#5258]
**Issue type:**

[x] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
I don't know if is a bug or not but when i perform an update it selects the entity again.
Is it possible to avoid this second select?

Example code:
```
let example = await Example.findOne({ id: "ID_HERE" });
if (!example)
    throw new Error("Not found");

example.data = 'NEW_DATA';
example.save();
```
Result Query:
```
query: SELECT "Example"."id" AS "Example_id", "Example"."data" AS "Example_data" FROM "example" "Example" WHERE "Example"."id" = $1 LIMIT 1 -- PARAMETERS: ["104d691c-37a3-4eac-9185-17cb987d938a"]
query: SELECT "Example"."id" AS "Example_id", "Example"."data" AS "Example_data" FROM "example" "Example" WHERE "Example"."id" IN ($1) -- PARAMETERS: ["104d691c-37a3-4eac-9185-17cb987d938a"]
query: START TRANSACTION
query: UPDATE "example" SET "data" = $2 WHERE "id" IN ($1) -- PARAMETERS: ["104d691c-37a3-4eac-9185-17cb987d938a","EXAMPLE DATA"]
query: COMMIT
```

**Comment by michaelbromley** [comment]
Hi,

Check the docs here: https://typeorm.io/#/repository-api/additional-options

You can pass an options object and set `reload: false` to prevent the additional SELECT after the UPDATE.

**Comment by angelvega93** [comment]
michaelbromley, the select is before UPDATE, i already tried the code
`example.save({ reload: false });`

But it does not work, it keeps selecting before UPDATE.

**Comment by michaelbromley** [comment]
Ah I see. AFAIK this extra SELECT statement is related to the `.findOne()` method. If you only perform the findOne without the `.save()`, you should see the same two SELECT queries, so the issue is independent from the save method.

My understanding is that this way of doing 2 SELECTS is related to the way TypeORM handles the `take` property of the FindManyOptions. It needs to do two queries in order to correctly handle `take` on joined tables. Internally, `findOne()` is using a similar code path but pre-filling `take: 1`:

https://github.com/typeorm/typeorm/blob/95c27c3b9ed3defd225922f3be106307f5663f07/src/entity-manager/EntityManager.ts#L842-L847

I ran into this issue myself when trying to optimize some perf issues and ended up using the QueryBuilder API directly on critical queries to get around this issue.

**Comment by angelvega93** [comment]
No, if i do not use the `.save()` method it just does one SELECT.

I also tried with MySQL, the behavior is the same, it SELECTS before UPDATE, maybe this is a bug, because it should be after the update, and in case of PostgreSQL it should use RETURNING.

**Comment by Mati365** [comment]
any updates?

**Comment by angelvega93** [comment]
Check MikroORM

**Comment by Mati365** [comment]
@angelvega93 MikroORM MikroORM sucks (even more than TypeORM) xD It has less than half of TypeORM features and has it's easier to shoot yourself in the foot due to memory leaks and race conditions

**Comment by samyan** [comment]
The same here, working with Mysql. Any solution?

**Comment by DillonSadofsky** [comment]
I don't believe this is a bug at all, though I can't speak for the devs.  From what I can tell you can pass a partial object to an EntityManager.save call.  IE, just the primary key and maybe a couple fields to update.  Internally the framework grabs the current version of the entity from the repository before building an UPDATE so it can compare your properties to the current values in the table.  It seems to do this so that it can build a minimum UPDATE statement (only changed fields) or skip it entirely if all values match.

I can see how that would conflict with many people's expectations of skipping another round trip, but I guess the way of thinking about it is that TypeORM  has the opinion that its better to do another round trip for a read if it means skipping getting row level (if InnoDB, or table level if the engine is MyISAM) locks in MySQL and etc.  This is semi-reasonable.

I believe the {reload: false} is *supposed* to control whether a fresh version of an object is returned afterwards, and I think (possibly a bug) it is only used on INSERTs.  By default if you do an INSERT with a partial object, it has to fully reload the object after save to get various fields that were set by column defaults or triggers.  I think reload: false turns this off and you have to reselect the current row yourself if you want it.

I think though that there is an issue where this same behavior is not working on updates (IE the reload AFTER the update, not the one that is BEFORE the update) and reload true or false has no effect on that.

**Comment by dberardo-com** [comment]
is there any other way to insert without select  using save ? the problem with the normal "insert" query is that insert does not cascade subobjects etc ... what is the alternative then ?

**Comment by dberardo-com** [comment]
still relevant
